.github/workflows/ci.yml

```diff
@@ -8,166 +8,160 @@ on:
-    # - name: Check WebGL
-    #  run: cargo check --all-targets --target=wasm32-unknown-unknown --features webgl
-
   build:
-    name: ${{ matrix.name }}
-    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-10.15, ubuntu-20.04, windows-2019]
-        channel: [stable, nightly]
         include:
-          - name: MacOS Stable
+          # Windows
+          - name: Windows x86_64
+            os: windows-2019
             channel: stable
+            target: x86_64-pc-windows-msvc
+            kind: compile
+          
+          # MacOS
+
+          # Mac has no software runners, so don't run tests
+          - name: MacOS x86_64
             os: macos-10.15
-            prepare_command:
-            clippy_params:
-            additional_core_features:
-            additional_player_features: winit
-            wrapper_features: trace
-          - name: MacOS Nightly
+            channel: stable
+            target: x86_64-apple-darwin
+            kind: compile
+
+          
+          # IOS
+          - name: IOS Nightly x86_64
             os: macos-10.15
             channel: nightly
-            prepare_command:
-            clippy_params:
-            additional_core_features:
-            additional_player_features:
-            wrapper_features:
-          - name: Ubuntu Stable
+            target: aarch64-apple-ios
+            kind: compile
+
+
+          # Linux
+          - name: Linux x86_64
             os: ubuntu-20.04
             channel: stable
-            prepare_command:
-            clippy_params: --examples --all
-            additional_core_features: trace,serial-pass
-            additional_player_features: winit
-            wrapper_features: spirv,replay
-          - name: Ubuntu Nightly
+            target: x86_64-unknown-linux-gnu
+            kind: test
+
+          - name: Linux Nightly x86_64
             os: ubuntu-20.04
             channel: nightly
-            prepare_command: |
-              echo "Installing Vulkan"
-              sudo apt-get update -y -qq
-              sudo add-apt-repository ppa:ubuntu-x-swat/updates -y
-              sudo apt-get update
-              sudo apt install -y libxcb-xfixes0-dev mesa-vulkan-drivers
-            clippy_params:
-            additional_core_features: serial-pass
-            additional_player_features:
-            wrapper_features:
-          - name: Windows Stable
-            os: windows-2019
-            channel: stable
-            prepare_command: rustup default stable-msvc
-            clippy_params:
-            additional_core_features:
-            additional_player_features:
-            wrapper_features:
-          - name: Windows Nightly
-            os: windows-2019
+            target: x86_64-unknown-linux-gnu
+            kind: compile
+
+          
+          # Android
+          - name: Android aarch64
+            os: ubuntu-20.04
+            channel: nightly
```

Comment:
Similarly I think we'd want to target Android and wasm on stable instead of nightly